### ISSUE TITLE
lack finetune and linprobe code

### ISSUE THREAD

**Issue by verazuo** [Sara-Ahmed/SiT#28]
Hi,

Thanks for sharing this wonderful project.
When I run the following commands to finetune the project, I find that some code for finetuning and linprobing is lacking, which lead following commands cannot run at all.
```
python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --batch-size 120 --epochs 501 --min-lr 5e-6 --training-mode 'finetune' --data-set 'STL10' --finetune 'checkpoints/SSL/STL10/checkpoint.pth' --output 'checkpoints/finetune/STL10' --validate-every 10
```

Also, many function snippets in `utils.py` are missed, i.e., `utils.restart_from_checkpoint`, `utils.fix_random_seeds(args.seed)`, also `utils.get_sha()`.

Would you please share it?

Thank you.

Best,
Vera

**Comment by Sara-Ahmed** [comment]
Thanks for your interest in our work. We did some modifications, and some files are still missing in the github. I will soon add the files and update the readme file and update you.

**Comment by verazuo** [comment]
Thanks for your quick response.

I noticed some previous commit contains the fine-tune code, e.g. https://github.com/Sara-Ahmed/SiT/tree/1aacd6adcd39b71efc903d16b4e9095b97dda76f

Is it possible for me to only run some lin-probing tasks based on this commit and the released checkpoint?

**Comment by MoMarky** [comment]
Hi, 
Thanks for sharing this wonderful project and I'm also very interested in your work.
But it seems that some function codes in utils.py are still missing.
Would you upload these codes soon?

Thank you,

Best,
Mark

**Comment by Sara-Ahmed** [comment]
Sorry for the late response. the utils file is updated.
